spec/lib/api/hub_config_api_spec.rb

```diff
@@ -0,0 +1,121 @@
+require 'spec_helper'
+require 'rails_helper'
+require 'cgi'
+
+describe HubConfigApi do
+  let(:hub_config_api) { HubConfigApi.new }
+  let(:entity_id) { 'http://www.test-rp.gov.uk/SAML2/MD' }
+  let(:wrong_entity_id) { 'wrong-entity-id' }
+  let(:encryption_certificate) { 'base64-x509-encryption-cert' }
+  let(:signing_certificate_one) { 'base64-x509-signing-cert-one' }
+  let(:signing_certificate_two) { 'base64-x509-encryption-cert-two' }
+  let(:hub_response_for_encryption) { 
+    {
+      issuerId: entity_id,
+      certificate: encryption_certificate,
+      keyUse: 'Encryption',
+      federationEntityType: 'RP',
+    }
+  }
+
+  let(:hub_response_for_signing) { 
+    [
+      {
+        issuerId: entity_id,
+        certificate: signing_certificate_one,
+        keyUse: 'Signing',
+        federationEntityType: 'RP',
+      }
+    ]
+  }
+
+  let(:hub_response_for_signing_when_dual_running) { 
+    [
+      {
+        issuerId: entity_id,
+        certificate: signing_certificate_one,
+        keyUse: 'Signing',
+        federationEntityType: 'RP',
+      },
+      {
+        issuerId: entity_id,
+        certificate: signing_certificate_two,
+        keyUse: 'Signing',
+        federationEntityType: 'RP',
+      }
+    ]
+  }
+
+  describe '#healthcheck' do
+    it 'should return service-status from healtheck' do
```

Comment:
This is fine but it 'returns... is better, it'should  implies it might not..

Reply:
I would disagree that it implies something else... I think it's clear and I'm just being consistent with the rest of the programme (i.e. 441 tests using `should` in verify-frontend and 33 tests in this repo alone)

Reply:
in my opinion it reads better when you have failing tests.. what it should be doing while it's not